kafka.go

```diff
@@ -213,13 +213,34 @@ func (client *KafkaClient) Publish(publishBuilder *PublishBuilder) error {
 
 // Publish send event to a topic
 func (client *KafkaClient) publishEvent(ctx context.Context, topic, eventName string, config kafka.WriterConfig,
-	message kafka.Message) error {
+	message kafka.Message) (err error) {
+	var writer *kafka.Writer
+	
+	defer func() {
+		if r := recover(); r != nil {
+			logrus.Warnf("unable to publish event to: %s. attempt to recover", topic)
+
+			if writer == nil {
+				err = errors.New("writer is nil")
+				return
+			}
+
+			if err = writer.Close(); err != nil {
+				logrus.Errorf("unable to publish event to: %s, unable to close writer: %s", topic, err)
```

Comment:
you dont need to change the 2nd "%v" to "%s" because the value is `err`
if you want to use "%s", the value should be "err.Error()" because it's string